app/routes/__index/certificate/index.tsx

```diff
@@ -1,11 +1,83 @@
+    validFrom: currentDate,
+    validTo: new Date(currentDate.getFullYear(), currentDate.getMonth() + 6),
+  };
+
+  if (loading) {
+    return <Loading />;
+  }
+
   return (
-    <div>
-      <Heading as="h1" size="3xl">
-        Certificate Info or option to request a certificate
-      </Heading>
-    </div>
+    <Center>
+      <Flex flexDirection="column" gap="5" width="4xl">
+        {certificateRequested && (
+          <>
+            <Description
+              description="Certificate: Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s"
+              certRequested={true}
+              validFrom={certificate.validFrom}
+              validTo={certificate.validTo}
+            />
+            <Divider />
+            <CertificateDisplay
+              title="Public Key"
+              value={certificate.certificate}
+              description="Public: Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s"
+            />
+            <Divider />
+            <CertificateDisplay
+              title="Private Key"
+              value={certificate.privateKey}
+              description="Private: Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s"
+            />
+          </>
+        )}
+        {!certificateRequested && (
```

Comment:
This pattern feels off to me.  We have a state that will result in one of two component sub-trees being rendered in this view.  I think you should make two components and do:

```ts
{ certificateRequested ? <View1 /> : <View2 /> }
```

Maybe name them:

1. CertificateRequested
2. CertificateAvailable

There might be something better, not sure.

Reply:
I actually thought about doing this on my previous commits, but I can't seem to figure out how I'll handle the button for `request certificate`

Reply:
oh it's actually fairly simple to do...